framework/x509.py

```diff
@@ -54,16 +54,15 @@ def __init__(self, cert_path=None, key_path=None, default=False):
 
     @staticmethod
     def __write(path, data):
-        fdesc = open(path, "wt")
-        fdesc.write(data.decode())
-        fdesc.close()
+        with open(path, 'wt') as fdesc:
+            fdesc.write(data.decode())
 
     def __encoding(self):
         if self.format == 'pem':
             return serialization.Encoding.PEM
-        else:
-            raise NotImplementedError("Not implemented encoding: %s"
-                                      % self.format)
+        raise NotImplementedError(
+            'Not implemented encoding: {0}'.format(self.format),
```

Comment:
The string
`f'Not implemented key algorithm: {self.key_alg}',`
is more readable to me